### DISCUSSION TITLE
502 Bad Gateway

### DISCUSSION THREAD

**Discussion by huwgit** [vendurehq/vendure#1976] [Q&A]
Not vendure specific I know. Underneath I get a message nginx.1,18.0. 

My host seems to think it is because I am using certbot for https and its a certificate issue. I restart the website using pm2. Another thing on mobile my images don't always load properly - I am using Remix. Any advice appreciated.

**Comment by michaelbromley** [comment]
Hi,
Maybe I can assist if you share more details, such as

- nginx config
- vendure apiOptions config
- pm2 config

and anything else you think might be relevant.

**Reply by huwgit** [reply]
It runs fine for 3/4 days then I restart. Here is the nginx:

server {

  server_name abnstone.co.uk www.abnstone.co.uk;

  location / {

    proxy_pass http://localhost:3001;

    proxy_http_version 1.1;

    proxy_set_header Upgrade $http_upgrade;

    proxy_set_header Connection 'upgrade';

    proxy_set_header Host $host;

    proxy_cache_bypass $http_upgrade;

  }
  location /admin {

    proxy_pass http://localhost:3000;

    proxy_http_version 1.1;

    proxy_set_header Upgrade $http_upgrade;

    proxy_set_header Connection 'upgrade';

    proxy_set_header Host $host;

    proxy_cache_bypass $http_upgrade;

  }
  location /assets {

    proxy_pass http://localhost:3000;

    proxy_http_version 1.1;

    proxy_set_header Upgrade $http_upgrade;

    proxy_set_header Connection 'upgrade';

    proxy_set_header Host $host;

    proxy_cache_bypass $http_upgrade;

  }

  location /shop-api {

    proxy_pass http://localhost:3000;

    proxy_http_version 1.1;

    proxy_set_header Upgrade $http_upgrade;

    proxy_set_header Connection 'upgrade';

    proxy_set_header Host $host;

    proxy_cache_bypass $http_upgrade;

  }

  location /health {

    proxy_pass http://localhost:3000;

    proxy_http_version 1.1;

    proxy_set_header Upgrade $http_upgrade;

    proxy_set_header Connection 'upgrade';

    proxy_set_header Host $host;

    proxy_cache_bypass $http_upgrade;

  }

    listen [::]:443 ssl ipv6only=on; # managed by Certbot
    listen 443 ssl; # managed by Certbot
    ssl_certificate /etc/letsencrypt/live/abnstone.co.uk/fullchain.pem; # managed by Certbot
    ssl_certificate_key /etc/letsencrypt/live/abnstone.co.uk/privkey.pem; # managed by Certbot
    include /etc/letsencrypt/options-ssl-nginx.conf; # managed by Certbot
    ssl_dhparam /etc/letsencrypt/ssl-dhparams.pem; # managed by Certbot

}
server {
    if ($host = www.abnstone.co.uk) {
        return 301 https://$host$request_uri;
    } # managed by Certbot


    if ($host = abnstone.co.uk) {
        return 301 https://$host$request_uri;
    } # managed by Certbot

  listen 80;

  listen [::]:80;

  server_name abnstone.co.uk www.abnstone.co.uk;
    return 404; # managed by Certbot

}

**Reply by huwgit** [reply]
and vendure.config:

import {
    dummyPaymentHandler,
    DefaultJobQueuePlugin,
    DefaultSearchPlugin,
    VendureConfig,
} from '@vendure/core';
import { LanguageCode } from '@vendure/core';
import { defaultEmailHandlers, EmailPlugin } from '@vendure/email-plugin';
import { AssetServerPlugin } from '@vendure/asset-server-plugin';
import { AdminUiPlugin } from '@vendure/admin-ui-plugin';
import 'dotenv/config';
import path from 'path';
import { StripePlugin } from '@vendure/payments-plugin/package/stripe';

const IS_DEV = process.env.APP_ENV === 'dev';

export const config: VendureConfig = {
    apiOptions: {
        port: 3000,
        adminApiPath: 'admin-api',
        shopApiPath: 'shop-api',
        // The following options are useful in development mode,
        // but are best turned off for production for security
        // reasons.
        ...(IS_DEV ? {
            adminApiPlayground: {
                settings: { 'request.credentials': 'include' } as any,
            },
            adminApiDebug: true,
            shopApiPlayground: {
                settings: { 'request.credentials': 'include' } as any,
            },
            shopApiDebug: true,
        } : {}),
    },
    authOptions: {
        tokenMethod: ['bearer', 'cookie'],
        requireVerification: true,
        superadminCredentials: {
            identifier: process.env.SUPERADMIN_USERNAME,
            password: process.env.SUPERADMIN_PASSWORD,
        },
        cookieOptions: {
          secret: process.env.COOKIE_SECRET,
        },
    },
    dbConnectionOptions: {
        type: 'better-sqlite3',
        // See the README.md "Migrations" section for an explanation of
        // the `synchronize` and `migrations` options.
        synchronize: true,
        migrations: [path.join(__dirname, './migrations/*.+(js|ts)')],
        logging: false,

        database: path.join(__dirname, '../vendure.sqlite'),
    },
    paymentOptions: {
        paymentMethodHandlers: [dummyPaymentHandler],
    },
    // When adding or altering custom field definitions, the database will
    // need to be updated. See the "Migrations" section in README.md.
    customFields: {
                    Product: [
            { name: 'weight', type: 'string'},
            { name: 'thickness', type: 'string'},
            { name: 'size', type: 'localeString',
            label: [
                { languageCode: LanguageCode.en, value: 'Dimensions LxWxT' },
              ], },
            { name: 'area', type: 'string',
            label: [
                { languageCode: LanguageCode.en, value: 'Tile area' },
              ],},
            { name: 'boxarea', type: 'string',
            label: [
                { languageCode: LanguageCode.en, value: 'Box area' },
              ],},
            { name: 'pricem2', type: 'string',
            label: [
                { languageCode: LanguageCode.en, value: 'Price per SQM' },
              ],},
              { name: 'pricebox', type: 'string',
              label: [
                  { languageCode: LanguageCode.en, value: 'Price per Box' },
                ],},

            { name: 'material', type: 'string',
            options: [
                    { value: 'Product' },
                { value: 'Granite' },
                { value: 'Marble' },
                { value: 'Quartz' },
                { value: 'Porcelain' },
                { value: 'Ceramic' },
              ]  },
            { name: 'finish', type: 'string',
            options: [
                { value: 'Matt' },
                { value: 'Polish' },
                { value: 'Gloss' },
              ] },
              { name: 'sample', type: 'string',
              options: [
                  { value: 'Not available' },
                  { value: 'Available' },

                ] },
                                                     
                { name: 'soldby', type: 'string',  defaultValue: 'per tile',
                options: [
                    { value: 'per tile' },
                    { value: 'per box' },
                    { value: 'per SQM' },

                  ] },
          ]
    },
    plugins: [
              StripePlugin.init({
    apiKey: "xxx",
    webhookSigningSecret: "xxx",
    // This prevents different customers from using the same PaymentIntent
    storeCustomersInStripe: true,
  }),
        AssetServerPlugin.init({
            route: 'assets',
            assetUploadDir: path.join(__dirname, '../static/assets'),
            // For local dev, the correct value for assetUrlPrefix should
            // be guessed correctly, but for production it will usually need
            // to be set manually to match your production url.
            assetUrlPrefix: IS_DEV ? undefined : 'https://www.my-shop.com/assets',
        }),
        DefaultJobQueuePlugin.init({ useDatabaseForBuffer: true }),
        DefaultSearchPlugin.init({ bufferUpdates: false, indexStockStatus: true }),
        EmailPlugin.init({
            devMode: true,
            outputPath: path.join(__dirname, '../static/email/test-emails'),
            route: 'mailbox',
            handlers: defaultEmailHandlers,
            templatePath: path.join(__dirname, '../static/email/templates'),
            globalTemplateVars: {
                // The following variables will change depending on your storefront implementation.
                // Here we are assuming a storefront running at http://localhost:8080.
                fromAddress: '"example" <noreply@example.com>',
                verifyEmailAddressUrl: 'http://localhost:8080/verify',
                passwordResetUrl: 'http://localhost:8080/password-reset',
                changeEmailAddressUrl: 'http://localhost:8080/verify-email-address-change'
            },
        }),
        AdminUiPlugin.init({
            route: 'admin',
            port: 3002,
        }),
    ],
};

**Reply by michaelbromley** [reply]
Thanks for the extra info.

So when it gives the 502 response, have you checked to see if the Vendure server is still running? If not, is there any error in the server logs?

**Reply by huwgit** [reply]
Yes the vendure server is still running.

**Reply by michaelbromley** [reply]
ok. I'm not sure what's going on then. As you say, it's not likely to be Vendure-specific so maybe you can get some assistance from a general forum like StackOverflow?
